### PR TITLE
Ecdc 3463 migrate to confluent kafka python

### PR DESCRIPTION
**Pull request by amues**

### CODE REVIEW

File: wherefore/KafkaConsumer.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+    def offsets_for_times(self, partitions_and_times: Dict[int, int]) -> Dict[int, int]:
+        """
+        Get offsets per partition from timestamps in ms for assigned topic.
+
+        :param partitions_and_times: Dict with partitions and timestamps in ms.
+        :return: Dict with partitions and offsets (None if timestamp is after last message).
+        """
+        try:
+            a = self._consumer.assignment()
```
Comment (mattclarke) [inline wherefore/KafkaConsumer.py]: I don't think we should be using single character names for things which have wide scope
```suggestion
            assigned_partitions = self._consumer.assignment()
```

File: wherefore/KafkaConsumer.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        :param topic: Topic name.
+        :param partition: Partition.
+        :return: List of messages.
+        """
+        try:
+            msgs = self._consumer.consume(max_msgs, timeout=self._timeout_s)
+        except KafkaException as e:
+            self._raise_from_kafka_exception(e)
+
+        r = []
```
Comment (mattclarke) [inline wherefore/KafkaConsumer.py]:
```suggestion
        result = []
```

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      cd ${pipeline_builder.project}
+      python -m black --check .
+    """
+  } // stage
+
+  pipeline_builder.stage("${container.key}: Static Analysis") {
+    // E203 formatting handled by black
+    // E501 formatting handled by black
+    // W503 complains about splitting if across lines which conflicts with black
+    // W605 invalid escape sequence '\d'
```
Comment (mattclarke) [inline Jenkinsfile]: These "ignores" could be put in a `.flake8` file rather than polluting the jenkinsfile

File: test/message_generator.py [hunk 4]
```diff
@@ -11,7 +11,7 @@
     serialise_f144,
     serialise_hs00,
     serialise_hs01,
-    serialise_ndar,
+    # serialise_ndar,
```
Comment (mattclarke) [inline test/message_generator.py]: Is there a reason this was commented out rather than deleted?

**Review by mattclarke — APPROVED** [review]

File: test/message_generator.py (hunk 4 above)
Comment (amues) [reply test/message_generator.py]: No, I just wasn't sure initially why it wasn't being used, I'll remove it.

**Pull request merged**